logsmanagement/helper.h

```diff
@@ -0,0 +1,330 @@
+ */
+static inline uint64_t get_unix_time_ms() {
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    uint64_t s1 = (uint64_t)(tv.tv_sec) * 1000;
+    uint64_t s2 = (uint64_t)(tv.tv_usec) / 1000;
+    return s1 + s2;
+}
+
+/**
+ * @brief Extract file_basename from full file path
+ * @param path String containing the full path.
+ * @return Pointer to the file_basename string
+ */
+static inline char *get_basename(const char *const path) {
+    if(!path) return NULL;
+    char *s = strrchr(path, '/');
+    if (!s)
+        return strdupz(path);
+    else
+        return strdupz(s + 1);
+}
+
+#ifdef LOGS_MANAG_DEBUG_LEV
+/**
+ * @brief Custom formatted print implementation
+ */
+static inline void fprintf_log_internal(Log_level log_level, FILE *stream, const char *format, ...) {
+    if (log_level > LOGS_MANAG_DEBUG_LEV) 
+        return; 
+
+    const char info_msg_prefix[] = "LOGS_MANAG_INFO: ";
+    const char warn_msg_prefix[] = "LOGS_MANAG_WARN: ";
+    const char error_msg_prefix[] = "LOGS_MANAG_ERROR: ";
+
+    const int msg_len = MAX_LOGS_MANAG_LOG_MSG_LEN;
+    char msg[msg_len];
+
+    va_list args;
+    va_start(args, format);
+
+    switch (log_level) {
+        case LOGS_MANAG_DEBUG:
+            fprintf(stream, "LOGS_MANAG_DEBUG: ");
+            vfprintf(stream, format, args);
+            break;
+        case LOGS_MANAG_INFO:
+            sprintf(msg, info_msg_prefix);
+            vsnprintf(&msg[sizeof(info_msg_prefix) - 1], msg_len - (sizeof(info_msg_prefix) - 1), format, args);
+            info("%s", msg);
+            break;
+        case LOGS_MANAG_WARNING:
+            sprintf(msg, warn_msg_prefix);
+            vsnprintf(&msg[sizeof(warn_msg_prefix) - 1], msg_len - (sizeof(warn_msg_prefix) - 1), format, args);
+            info("%s", msg);
+            break;
+        case LOGS_MANAG_ERROR:
+            sprintf(msg, error_msg_prefix);
+            vsnprintf(&msg[sizeof(error_msg_prefix) - 1], msg_len - (sizeof(error_msg_prefix) - 1), format, args);
+            error("%s", msg);
+            break;
+        default:
+            break;
+    }
+
+    va_end(args);
+}
+#endif  // LOGS_MANAG_DEBUG_LEV
+
+typedef enum {
+    STR2XX_SUCCESS = 0,
+    STR2XX_OVERFLOW,
+    STR2XX_UNDERFLOW,
+    STR2XX_INCONVERTIBLE
+} str2xx_errno;
+
+/* Convert string s to int out.
+ * https://stackoverflow.com/questions/7021725/how-to-convert-a-string-to-integer-in-c
+ *
+ * @param[out] out The converted int. Cannot be NULL.
+ * @param[in] s Input string to be converted.
+ *
+ *     The format is the same as strtol,
+ *     except that the following are inconvertible:
+ *     - empty string
+ *     - leading whitespace
+ *     - any trailing characters that are not part of the number
+ *     Cannot be NULL.
+ *
+ * @param[in] base Base to interpret string in. Same range as strtol (2 to 36).
+ * @return Indicates if the operation succeeded, or why it failed.
+ */
+static inline str2xx_errno str2int(int *out, char *s, int base) {
```

Comment:
Unless I am missing something, this function is also not necessary, please, take a look at:

```sh
libnetdata/inlined.h:#define str2kernel_uint_t(string) str2uint32_t(string)
libnetdata/inlined.h:#define str2kernel_uint_t(string) str2uint64_t(string)
libnetdata/inlined.h:#define str2pid_t(string) str2uint32_t(string)
libnetdata/inlined.h:static inline int str2i(const char *s) {
libnetdata/inlined.h:static inline long str2l(const char *s) {
libnetdata/inlined.h:static inline uint32_t str2uint32_t(const char *s) {
libnetdata/inlined.h:static inline uint64_t str2uint64_t(const char *s) {
libnetdata/inlined.h:static inline unsigned long str2ul(const char *s) {
libnetdata/inlined.h:static inline unsigned long long str2ull(const char *s) {
libnetdata/inlined.h:static inline long long str2ll(const char *s, char **endptr) {
libnetdata/inlined.h:    *result = str2ull(buffer);
```

Reply:
@Dim-P , please, take a look in previous comment

Reply:
Well, I am returning here again, please, take a look here.